Add documentation for new .NET 8 diagnostic configuration (#38012)
dotnet/docs: docs/core/tools/dotnet-environment-variables.md

When set to `1`, enables debugging, profiling, and other diagnostics via the [Diagnostic Port](../diagnostics/diagnostic-port.md). Defaults to 1.

### `DOTNET_EnableDiagnostics_IPC`

> > [!NOTE]
> > Available since .NET 8
> When set to `1`, enables the [Diagnostic Port](../diagnostics/diagnostic-port.md). Defaults to 1.

Reviewer: ```suggestion
Starting with .NET 8, when set to `1`, enables the [Diagnostic Port](../diagnostics/diagnostic-port.md). Defaults to 1.
```

### `DOTNET_EnableDiagnostics_Debugger`

> > [!NOTE]
> > Available since .NET 8
> When set to `1`, enables debugging. Defaults to 1.

Reviewer: ```suggestion
Starting with .NET 8, when set to `1`, enables debugging. Defaults to 1.
```

### `DOTNET_EnableDiagnostics_Profiler`

> > [!NOTE]
> > Available since .NET 8
> When set to `1`, enables profiling. Defaults to 1.

Reviewer: ```suggestion
Starting with .NET 8, when set to `1`, enables profiling. Defaults to 1.
```
